doc/usecases/core/UC_keyset_create.md

```diff
@@ -0,0 +1,26 @@
+# Use Case: Create `KeySet`
+
+## Summary
+
+- **Title:** Create `KeySet`
+- **Scope:** `libelektra-core`
+- **Level:** Developer Goal
+- **Actors:** Core, Caller
+- **Brief:** Caller creates new `KeySet` instance
+
+## Scenarios
+
+- **Precondition:** -
+- **Main success scenario:**
+  - Caller requests to create a new `KeySet`
+  - Core instantiates an emtpy `KeySet` and returns it
+- **Alternative scenario:**
+  - Caller requests to create a new `KeySet`, with known content
+  - Core instantiates an appropriately sized `KeySet`, fills it and returns it
```

Comment:
I was intentionally vague here. I did not want to require this a single `ksNew` function that takes everything as arguments. Instead I wanted to still allow using multiple function calls to achieve this, e.g.

```c
KeySet * ks = ksCreate (15);
ksInsert (ks, key1);
ksInsert (ks, key2);
// ...

// OR

KeySet * ks = ksInsert (ksInsert (ksCreate (2), key1), key2);

// OR

KeySet * ks = ksInsertAll (ksCreate (15), key1, key2, ...);
```

One thing I may still change, is that the `ksNew`-like function could exist outside `libelektra-core`.

Reply:
Use cases are also decisions. We decide here if we want this feature in the core.

The reason for the size argument was that the signature `ksNew(...)` with only `Key`s is not possible. So if we want to get rid of the use case to construct a `ksNew` in one go in the core, we should also get rid of this size.

Imho it is okay to have a `ksNew()` (without arguments) in the core. `ksCreate()` with the vaargs can be in an extra lib, it is not really minimal.

Reply:
IMHO this discussion is far too detailed already for this PR. I'll remove the alternative scenario and we'll discuss the details of creating keysets in #4244.